### PR TITLE
HS-1105: Update Active discovery 

### PR DESCRIPTION
**Pull request by cgorantla**
## Description
Update Active Discovery
 - BFF : adds location, tags , updated method names and it's parameters
 - Inventory: Move active discovery to use active_discovery_config table 

## Jira link(s)
- https://issues.opennms.org/browse/HS-1105
## Checklist
* [ ] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.

### CODE REVIEW

File: inventory/main/src/main/java/org/opennms/horizon/inventory/service/ActiveDiscoveryService.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        activeDiscoveryConfig = repository.save(activeDiscoveryConfig);
+        return mapper.modelToDto(activeDiscoveryConfig);
+    }
+
+    public List<ActiveDiscoveryDTO> listDiscoveryConfigs(String tenantId) {
+        var entities = repository.findByTenantId(tenantId);
+        return entities.stream().map(mapper::modelToDto).toList();
+    }
+
+    public Optional<ActiveDiscoveryDTO> getDiscoveryConfigByName(String name, String tenantId) {
```
Comment (thomasbigger584) [inline inventory/main/src/main/java/org/opennms/horizon/inventory/service/ActiveDiscoveryService.java:63]: is name unique? should it be ID instead?

**Review by BenjaminJ — COMMENTED** [review]
It's not easy to review since the files are in main/src/main,
but it looks okay.

File: inventory/main/src/main/java/org/opennms/horizon/inventory/service/ActiveDiscoveryService.java (hunk 1 above)
Comment (cgorantla) [reply inventory/main/src/main/java/org/opennms/horizon/inventory/service/ActiveDiscoveryService.java:63]: yeah, name may not be unique, will update this

**Review by thomasbigger584 — APPROVED** [review]

**Comment by cgorantla** [comment]
graphql request:

```
mutation { 
    createActiveDiscovery( 
        request: { 
            location: "minion-standalone-loc",
            configName : "myHome",
            ipAddresses : ["192.168.1.1"],
            snmpConfig : {
              readCommunities : ["public"],
              ports : [1162]
            }
        }
    ) {
        id,
        location,
        configName,
        ipAddresses,
        snmpConfig {
          ports,
          readCommunities
        }
    }
}
```
Response:

{
  "data": {
    "createActiveDiscovery": {
      "id": 4,
      "location": "minion-standalone-loc",
      "configName": "myHome",
      "ipAddresses": [
        "192.168.1.1"
      ],
      "snmpConfig": {
        "ports": [
          1162
        ],
        "readCommunities": [
          "public"
        ]
      }
    }
  }
}

**Pull request merged**